pkg/util/bandwidth/utils_test.go

```diff
@@ -30,41 +30,72 @@ func TestExtractPodBandwidthResources(t *testing.T) {
 	ten, _ := resource.ParseQuantity("10M")
 	twenty, _ := resource.ParseQuantity("20M")
 
-	testPod := func(ingress, egress string) *api.Pod {
+	testPod := func(ingressRate, ingressBurst, egressRate, egressBurst string) *api.Pod {
 		pod := &api.Pod{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{}}}
-		if len(ingress) != 0 {
-			pod.Annotations["kubernetes.io/ingress-bandwidth"] = ingress
+		if len(ingressRate) != 0 {
+			pod.Annotations["kubernetes.io/ingress-bandwidth"] = ingressRate
 		}
-		if len(egress) != 0 {
-			pod.Annotations["kubernetes.io/egress-bandwidth"] = egress
+		if len(ingressBurst) != 0 {
+			pod.Annotations["kubernetes.io/ingress-burst"] = ingressBurst
+		}
+		if len(egressRate) != 0 {
+			pod.Annotations["kubernetes.io/egress-bandwidth"] = egressRate
+		}
+		if len(egressBurst) != 0 {
+			pod.Annotations["kubernetes.io/egress-burst"] = egressBurst
 		}
 		return pod
 	}
 
 	tests := []struct {
 		pod             *api.Pod
-		expectedIngress *resource.Quantity
-		expectedEgress  *resource.Quantity
+		expectedIngress *BandwidthLimit
+		expectedEgress  *BandwidthLimit
 		expectError     bool
 	}{
 		{
 			pod: &api.Pod{},
 		},
 		{
-			pod:             testPod("10M", ""),
-			expectedIngress: &ten,
+			pod: testPod("10M", "", "10M", ""),
```

Comment:
you should add test cases with for `Burst`, happy cases, check that it errors if burst is set but rate not, ...

Reply:
done

Reply:
what about?
testPod("10M", "10M", "10M", "10M"),
testPod("10M", "10M", "10M", ""),
testPod("10M", "", "10M", "10M"),
testPod(lower than minBW, "10M", "10M", "10M") must err
testPod(greater than maxBW, "10M", "10M", "10M") must err
testPod("10M", lower than min burst, "10M", "10M") must err
testPod("10M", greater than max burst, "10M", "10M") must err

Reply:
that's precise and thorough enough, thanks for the advice, added some comment and updated, please review again.